iocage/cli/create.py

```diff
@@ -64,97 +62,108 @@ def validate_count(ctx, param, value):
-    "foo",
-    "bar"
-    ]
-}"""
-
-        if not os.path.isfile(pkglist):
-            ioc_common.logit({
-                "level"  : "EXCEPTION",
-                "message": f"{pkglist} does not exist!\n"
-                           "Please supply a JSON file with the format:"
-                           f" {_pkgformat}"
-            })
-        else:
-            try:
-                # Just try to open the JSON with the right key.
-                with open(pkglist, "r") as p:
-                    json.load(p)["pkgs"]  # noqa
-            except json.JSONDecodeError:
-                ioc_common.logit({
-                    "level"  : "EXCEPTION",
-                    "message": "Please supply a valid"
-                               f" JSON file with the format:{_pkgformat}"
-                })
-
-    if empty:
-        release = "EMPTY"
-
-    iocage = ioc.IOCage(skip_jails=True)
-
-    try:
-        iocage.create(release, props, count, pkglist=pkglist,
-                      template=template, short=short, _uuid=_uuid,
-                      basejail=basejail, empty=empty)
-    except RuntimeError as err:
-        if template:
-            # We want to list the available templates first
-            ioc_common.logit({
-                "level"  : "ERROR",
-                "message": f"Template: {release} not found!"
-            })
-            templates = ioc.IOCage().list("template")
-            for temp in templates:
-                ioc_common.logit({
-                    "level"  : "EXCEPTION",
-                    "message": f"Created Templates:\n  {temp[1]}"
-                })
+        jail_data["name"] = name
+
+    release = iocage.lib.Release.Release(
+        name=release,
+        logger=logger,
+        host=host,
+        zfs=zfs
+    )
+    if not release.fetched:
+        name = release.name
+        if not release.available:
+            logger.error(
+                f"The release '{release.name}' does not exist"
+            )
+            exit(1)
+
+        msg = (
+            f"The release '{release.name}' is available,"
+            "but not downloaded yet"
+        )
+        if no_fetch:
+            logger.error(msg)
             exit(1)
         else:
-            # Standard errors
-            ioc_common.logit({
-                "level"  : "EXCEPTION",
-                "message": err
-            })
+            logger.spam(msg)
+            logger.log("Automatically fetching release '{release.name}'")
+            release.fetch()
+
+    if basejail:
+        jail_data["basejail"] = True
+
+    if basejail_type is not None:
+        if not basejail:
+            logger.error(
+                "Cannot set --basejail-type without --basejail option")
+            exit(1)
+        jail_data["basejail_type"] = basejail_type
+
+    if props:
+        for prop in props:
+            try:
+                key, value = prop.split("=", maxsplit=1)
+                jail_data[key] = value
+            except:
```

Comment:
We probably want except (ValueError, KeyError) here. Catching KeyboardInterrupt does not seem good.